src/share/poudriere/common.sh

```diff
@@ -5090,6 +5090,9 @@ delete_old_pkg() {
 	fi
 
 	v="${pkgname##*-}"
+	# Strip the hash from the filename to leave just the version
+	v="${v%+*}"
```

Comment:
```
% pkg rquery %n-%v|grep -F +
afl++-2.62c
arpack++-2.3.0
arts++-1.1.a13_2,1
blitz++-1.0.2
bonnie++-1.98
bpl+-1.0_1
c++-gtk-utils-2.2.18
clisp-2.49.93+_1
crf++-0.54_1
cvs+ipv6-1.20120905_3
dbus-c++-0.9.1_5
dia2code+-1.0.0_7
dvd+rw-tools-7.1_3
exim-sa-exim-4.93.0.4+4.2.1_3
fc++-1.5
flvtool++-1.2.1_14
gmm++-5.3
golded+-1.1.5.a20170303
ja-timidity++-slang-2.15.0
lapack++-2.5.4_9
libcdio-paranoia-10.2+2.0.1
libodbc++-0.2.5_5
libsigc++-2.10.2
libsigc++12-1.2.7_2
libstdc++_stldoc_4.2.2-20071101_1
libxml++-2.40.1,1
linux-c7-libsigc++20-2.10.0
lucene++-3.0.7_14
make++-2.0
memtest86+-5.01_3
mpfrc++-3.6.2
mysql-connector-c++-1.1.13_1
ncbi-blast+-2.10.0
nurbs++-3.0.11_9
ocaml-camlp4-4.05+2
p5-Text-Format+NWrap-0.52_1
p5-Text-Tabs+Wrap-2013.0523_1
pcre++-0.9.5_5
png++-0.2.10
quickcheck++-0.0.3
s2tc-1.0+20151228_1
sdts++-1.5.1_19
smm++-6.1.1_1
timidity++-2.15.0
timidity++-emacs-2.15.0
timidity++-gtk-2.15.0
timidity++-motif-2.15.0
timidity++-slang-2.15.0
timidity++-xaw-2.15.0
timidity++-xskin-2.15.0
tintin++-2.02.02
tolua++-1.0.93_3
vera++-1.3.0_14
voro++-0.4.6_1
weblint++-1.15_3
wmweather+-2.17_1
xeyes+-2.02_2
xgalaga++-0.9
xlife++-2.0.1_5
yaz++-1.6.1_1
zh-tintin++-2.02.02
```

Reply:
The line above has already tripped off everything before the last -, so it is just %v, but there are still a few hits there:
2.49.93+_1
4.93.0.4+4.2.1_3
10.2+2.0.1
4.05+2
1.0+20151228_1

Reply:
As agreed on IRC, switched to using ~ instead of % or +